templates/canal.go

```diff
@@ -559,7 +559,7 @@ const CanalTemplateV113 = `
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1beta1
 metadata:
-  name: calico-node
+  name: calico
```

Comment:
What is the reason behind changing this from `calico` to `calico-node` and what is the impact of upgrading cluster with this change?

Reply:
The network addon job would fail with this error:
```
The ClusterRoleBinding "canal-calico" is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:"rbac.authorization.k8s.io", Kind:"ClusterRole", Name:"calico-node"}: cannot change roleRef
```
`roleRef` is an immutable part of the spec. Check https://github.com/kubernetes/kubernetes/issues/66353 and https://github.com/kubernetes/website/pull/11254 for details.

Reply:
got it